core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -151,7 +151,7 @@ protected void validate(TableMetadata base) {
       if (deletedDataFiles.size() > 0) {
         validateNoNewDeletesForDataFiles(
             base, startingSnapshotId, conflictDetectionFilter,
-            deletedDataFiles, caseSensitive);
+            deletedDataFiles, caseSensitive, false);
```

Comment:
It isn't clear what's happening here. If you're introducing a new method argument, can you add an override so that this doesn't need to change? If you think that the boolean is needed, then can you add an inline comment to explain what you're setting?